src/server/telemetry.js

```diff
@@ -0,0 +1,168 @@
+var http = require('http');
+var https = require('https');
+var _ = require('../utility');
+var urlHelpers = require('./telemetry/urlHelpers');
+
+var defaults = {
+  network: true,
+  networkResponseHeaders: false,
+  networkRequestHeaders: false,
+  log: true
+};
+
+function Instrumenter(options, telemeter, rollbar) {
+  this.options = options;
+  var autoInstrument = options.autoInstrument;
+  if (options.enabled === false || autoInstrument === false) {
+    this.autoInstrument = {};
+  } else {
+    if (!_.isType(autoInstrument, 'object')) {
+      autoInstrument = defaults;
+    }
+    this.autoInstrument = _.merge(defaults, autoInstrument);
```

Comment:
Nit: I think this would read more clearly as:

```
} else if (_.isType(autoInstrument, 'object')) {
  this.autoInstrument = _.merge(defaults, autoInstrument);
} else {
  this.autoInstrument = defaults;
}
```

Reply:
Almost. 

`_.merge()` returns a clone, so you want to take care to do:
`this.autoInstrument = _.merge(defaults); // acts as a deep clone`

The original version of this code does this, though the revised version makes it more apparent.

Reply:
I usually try to avoid if/else if possible, so to me, the original way is cleaner if it's doing the same thing. (it's just my preference - it seems to look cleaner, but I fine with both ways)

Reply:
Ah, right. So perhaps most self-evident:

```
} else if (_.isType(autoInstrument, 'object')) {
  this.autoInstrument = _.merge(defaults, autoInstrument);
} else {
  this.autoInstrument = _.cloneDeep(defaults);
}
```

Or, with @pawelsz-rb comment, perhaps this is the most self-documenting:

```
} else {
  this.autoInstrument = _.merge(defaults, _.isType(autoInstrument, 'object') ? autoInstrument : {});
}
```

In either case, I feel readability is enhanced and at the same time the extra not and the potential no-op of merging `defaults` with itself are removed.

Reply:
I think we might be bikeshedding a little bit, but first a couple things.

I enjoy using the ternary operator, but IMO that example isn't that readable. Maybe the nested function calls and multiple appearances of `autoInstrument` contribute to this.

I'm not opposed to `deepClone()` that calls `merge()` with one argument, but I'm not sure this PR is where to do it. `merge()` with a single argument is already used liberally throughout the code base. I'd rather address it at that level than a one-off here.